service/storage_proxy.cc

```diff
@@ -2005,6 +2006,11 @@ storage_proxy::get_paxos_participants(const sstring& ks_name, const dht::token &
  * @param tr_state trace state handle
  */
 future<> storage_proxy::mutate(std::vector<mutation> mutations, db::consistency_level cl, clock_type::time_point timeout, tracing::trace_state_ptr tr_state, service_permit permit, bool raw_counters) {
+    if (_cdc && _cdc->needs_cdc_augmentation(mutations)) {
+        return _cdc->augment_mutation_call(timeout, std::move(mutations)).then([this, cl, timeout, tr_state = std::move(tr_state), permit = std::move(permit), raw_counters](std::tuple<std::vector<mutation>, cdc::result_callback>&& t) mutable {
+            return _mutate_stage(this, std::move(std::get<0>(t)), cl, timeout, std::move(tr_state), std::move(permit), raw_counters);
+        });
+    }
```

Comment:
It looks a lot like what triggers infrastructure is doing in Cassandra, so may we should implement that and register cdc as a trigger.

Reply:
Maybe. But I would suggest leaving that as an enhancement, also because some chicken-egg stuff might get resolved with pavels upcoming changes etc.